dataactcore/models/baseInterface.py

```diff
@@ -157,3 +159,10 @@ def getNameFromDict(self, model, dictName, fieldName, fieldValue, idField):
                 return key
         # If not found, raise an exception
         raise ValueError("Value: " + str(fieldValue) + " not found in dict: " + str(dict))
+
+
+@contextmanager
+def databaseSession():
+    interface = BaseInterface()
+    yield interface.session
+    interface.close()
```

Comment:
Very nice, I like this approach.  For the call from the constructor, just make sure it happens only if self.session is None, otherwise we'll have a repeat of last week's issue with connections staying open.